app/js/vendor/asciidoctor.js

```diff
@@ -42656,7 +42656,7 @@ Opal.modules["asciidoctor/js/postscript"] = function(Opal) {
     Opal.const_set($nesting[0], 'LINE_CONTINUATION', " \\");
     Opal.const_set($nesting[0], 'LINE_CONTINUATION_LEGACY', " +");
     Opal.const_set($nesting[0], 'MATHJAX_VERSION', "2.7.4");
-    Opal.const_set($nesting[0], 'BLOCK_MATH_DELIMITERS', $hash2(["asciimath", "latexmath"], {"asciimath": ["\\$", "\\$"], "latexmath": ["\\[", "\\]"]}));
+    Opal.const_set($nesting[0], 'BLOCK_MATH_DELIMITERS', $hash2(["asciimath", "latexmath"], {"asciimath": ["\\$\\$", "\\$\\$"], "latexmath": ["\\[", "\\]"]}));
```

Comment:
I think it's unnecessary and the file will be overridden when we upgrade Asciidoctor.js
Could you please confirm that it's working without this change ?

Reply:
It won't work without that change.

Reply:
In order to make a piece of asciimath content centered on a clear line I need to do `data.math.root.display = "block"` on it.

We only want the **block** asciimath to be centered on a clear line. The inline asciimath should behave as before. Therefore we target only asciimath blocks. This happens in two steps:

1. Marking: asciimath blocks maked with `this.match['\\$\\$'].mode = ";mode=display";` (please notice the double dollar sign). Marking means applying `;mode=display`
If I work with the single dollar sign I will affect all asciimath content (inline and block), which is not intended.
2. Setting display block:
```
if (data.script.type.match(/;mode=display/)) {
  data.math.root.display = "block"
}
```
I understand now the workflow. We don't write to asciidoctor.js in this project, we only bring it unmodified from upstream.
I will undo the change on asciidoctor.js. However if we don't get the modification from upstream this change set is useless.
How should I proceed?

Reply:
Thanks for the detailed explanation. It makes sense :wink: 

> I understand now the workflow. We don't write to asciidoctor.js in this project, we only bring it unmodified from upstream.

You're absolutely right.

> I will undo the change on asciidoctor.js. However if we don't get the modification from upstream this change set is useless.

Asciidoctor.js is cross-compiled from Asciidoctor Ruby so upstream is really: https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/converter/html5.rb

That's why I was asking @mojavelinux's opinion on this issue because we don't want to have inconsistencies between extensions/projects (even if arguably the output is better with your changes)

> How should I proceed?

I think we first need to decide if the issue should be fixed in Asciidoctor Ruby or not.

Reply:
👍